### PR TITLE
[FIX] #129 - Wrap synchonize metadata method and give the user a warning

### PR DESCRIPTION
**Pull request by danimaribeiro**

### CODE REVIEW

**Comment by StefanRijnhart** [comment]
:+1:

File: magentoerpconnect/magento_model.py [hunk 1]
```diff
@@ -186,17 +187,23 @@ def check_magento_structure(self):
...
-                          'magento.store',
-                          'magento.storeview'):
-                # import directly, do not delay because this
-                # is a fast operation, a direct return is fine
-                # and it is simpler to import them sequentially
-                import_batch(session, model, backend.id)
-        return True
+        try:
+            session = ConnectorSession(self.env.cr, self.env.uid,
+                                       context=self.env.context)
```
Comment (guewen) [inline magentoerpconnect/magento_model.py:192]: Could you take the opportunity of this diff to improve that line to `ConnectorSession.from_env(self.env)`?

File: magentoerpconnect/magento_model.py [hunk 2]
```diff
@@ -186,17 +187,23 @@ def check_magento_structure(self):
...
+                    # import directly, do not delay because this
+                    # is a fast operation, a direct return is fine
+                    # and it is simpler to import them sequentially
+                    import_batch(session, model, backend.id)
+            return True
+        except Exception as e:
+            _logger.error(e.message, exc_info=True)
+            raise Warning(
+                _(u"Check your configuration, we can't get the data.\
+                Here is the error:\n%s") % str(e).decode('utf-8', 'ignore'))
```
Comment (guewen) [inline magentoerpconnect/magento_model.py:206]: Could you write it in the following manner:

``` python
raise Warning(
    _(u"Check your configuration, we can't get the data. "
      u"Here is the error:\n%s") % str(e).decode('utf-8', 'ignore'))
```

So we don't get blanks in the string?

**Comment by guewen** [comment]
Thanks for the contrib!

**Comment by StefanRijnhart** [comment]
Superseded by https://github.com/OCA/connector-magento/pull/231

**Pull request closed without merging**